packages/abi-coder/src/encoding/coders/EnumCoder.ts

```diff
@@ -36,7 +36,7 @@ export class EnumCoder<TCoders extends Record<string, Coder>> extends Coder<
   constructor(name: string, coders: TCoders) {
     const caseIndexCoder = new BigNumberCoder('u64');
     const encodedValueSize = Object.values(coders).reduce(
-      (max, coder) => Math.max(max, coder.encodedLength),
+      (min, coder) => Math.min(min, coder.encodedLength),
```

Comment:
Use the minimum for now, we can check the actual length later once we've decoded the case.